pandas/core/internals/blocks.py

```diff
@@ -2465,6 +2465,12 @@ def setitem(self, indexer, value):
                               klass=ObjectBlock,)
             return newb.setitem(indexer, value)
 
+    def equals(self, other):
+        # override for significant performance improvement
```

Comment:
why would u do this here and not in array_equivalent

Reply:
Because array_equivalent's docstring specifically says it assumes it is dealing with numpy arrays